client/post-editor/media-modal/detail/detail-preview-image.jsx

```diff
@@ -1,36 +1,77 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
+import React, { PropTypes, Component } from 'react';
+import { noop } from 'lodash';
+import classNames from 'classnames';
 
 /**
  * Internal dependencies
  */
 import ImagePreloader from 'components/image-preloader';
 import Spinner from 'components/spinner';
-import MediaUtils from 'lib/media/utils';
+import { url, isItemBeingUploaded } from 'lib/media/utils';
 
-export default React.createClass( {
-	displayName: 'EditorMediaModalDetailPreviewImage',
-
-	propTypes: {
+export default class EditorMediaModalDetailPreviewImage extends Component {
+	static propTypes = {
 		site: PropTypes.object,
-		item: PropTypes.object.isRequired
-	},
+		item: PropTypes.object.isRequired,
+	};
+
+	static defaultProps = {
+		onLoad: noop
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.onImagePreloaderLoad = this.onImagePreloaderLoad.bind( this );
+		this.state = { loading: true };
+	}
+
+	componentWillReceiveProps() {
+		this.setState( { loading: true } );
+	}
+
+	onImagePreloaderLoad() {
+		this.setState( { loading: false } );
+		this.props.onLoad();
+	}
 
 	render() {
-		const src = MediaUtils.url( this.props.item, {
+		const src = url( this.props.item, {
 			photon: this.props.site && ! this.props.site.is_private
 		} );
+		const uploading = isItemBeingUploaded( this.props.item );
+		const loading = this.state.loading;
+
+		const classes = classNames(
+			'editor-media-modal-detail__preview',
+			'is-image',
+			{ 'is-uploading': uploading },
+			{ 'is-loading': loading }
+		);
 
 		return (
-			<ImagePreloader
-				src={ src }
-				width={ this.props.item.width }
-				height={ this.props.item.height }
-				placeholder={ <Spinner /> }
-				alt={ this.props.item.alt || this.props.item.title }
-				className="editor-media-modal-detail__preview is-image" />
+			<div>
+				<img
+					src={ src }
+					width={ this.props.item.width }
+					height={ this.props.item.height }
+					alt={ this.props.item.alt || this.props.item.title }
+					className={ classes + ' is-fake' } />
+
+				<ImagePreloader
+					src={ src }
+					width={ this.props.item.width }
+					height={ this.props.item.height }
+					onLoad={ this.onImagePreloaderLoad }
+					placeholder={ <span /> }
+					alt={ this.props.item.alt || this.props.item.title }
+					className={ classes } />
+
+				{ ( loading || uploading ) && <Spinner counterClockwise={ uploading } /> }
```

Comment:
Let's stick with a single direction. I'm seeing a flash of clockwise rotation after an edit/restore.